src/test/java/HttpRequestBuilderTest.java

```diff
@@ -0,0 +1,40 @@
+import com.td.Mocks.MockHttpRequestReader;
+import com.td.Mocks.DummyIO;
+import com.td.HttpServer.HttpRequestBuilder;
+import com.td.HttpServer.HttpRequestParser;
+import com.td.HttpServer.HttpVerifier;
+import com.td.HttpServer.InvalidHttpRequestException;
+import java.io.IOException;
+
+public class HttpRequestBuilderTest extends junit.framework.TestCase {
+
+  HttpRequestParser parser;
+  MockHttpRequestReader mockReader;
+  HttpRequestBuilder builder;
+
+  protected void setUp() {
+    parser = new HttpRequestParser(new HttpVerifier());
+  }
+
+  public void testThatReaderGetBodyIsNotCalledIfNoContentLength() {
+    mockReader = new MockHttpRequestReader("GET / HTTP/1.1\n\n", new byte[0]);
+    builder = new HttpRequestBuilder(mockReader, parser);
+    try {
+    builder.createRequest(new DummyIO());
+    }
+    catch (InvalidHttpRequestException e) {}
```

Comment:
Probably. If a test method throws an exception that will fail the test. That's probably a fine way to fail a tests :) Not sure if everyone shares that opinion.